File: framework/scala/src/main/scala/soda/scala/unittest/WorkInput.scala
package soda.scala.unittest

import play.api.libs.json._

class WorkInput(jstr: String) {

  private val root = Json.parse(jstr)

  def id: Int = (root \ "id").get.as[Int]

  def hasExpected: Boolean = {
    val r = (root \ "expected")
    r.isDefined match {
      case false => false
      case true => r.get match {
        case null => false
        case JsNull => false
        case _ => true
      }
    }
  }

  def expected: JsValue = (root \ "expected").get

  def arg(index: Int): JsValue = (root \ "args" \ index).get

  def arguments: JsValue = (root \ "args").get

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/unittest/WorkInput.kt
package soda.kotlin.unittest

import kotlinx.serialization.json.*

class WorkInput(jstr: String) {

    private val root = Json.parseToJsonElement(jstr)

    val id: Int?
        get() = root.jsonObject["id"]?.jsonPrimitive?.int

    val hasExpected: Boolean
        get() = root.jsonObject["expected"]?.jsonPrimitive?.boolean ?: false

    val expected: JsonElement
        get() = root.jsonObject["expected"]!!

    fun arg(index: Int): JsonElement? = root.jsonObject["args"]?.jsonArray?.get(index)

    val arguments: JsonElement?
        get() = root.jsonObject["args"]

}